server/src/main/java/org/opensearch/indices/replication/SegmentReplicationTargetService.java

```diff
@@ -144,6 +148,55 @@ public SegmentReplicationTargetService(
         );
     }
 
+    @Override
+    protected void doStart() {
+        final ClusterService clusterService = indicesService.clusterService();
+        if (DiscoveryNode.isDataNode(clusterService.getSettings())) {
+            clusterService.addListener(this);
+        }
+    }
+
+    @Override
+    protected void doStop() {
+        final ClusterService clusterService = indicesService.clusterService();
+        if (DiscoveryNode.isDataNode(clusterService.getSettings())) {
+            indicesService.clusterService().removeListener(this);
```

Comment:
Use the `clusterService` variable here as well

Reply:
thanks missed both of these, updated.